File: BRACU ACM Student Chapter/Online Workshop on Programming Skills DAY 03 ( Java ) BRACU Data Structure/G.java

import java.util.ArrayList;
import java.util.Scanner;

public class G {

    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        int number = input.nextInt();

        while (number != 0) {

            if (number == 1) {
                System.out.println("Discarded cards:\nRemaining card: 1");
            } else {
                ArrayList<Integer> card = new ArrayList<>();

                for (int i = 0; i < number; i++) {
                    card.add(i + 1);
                }

                System.out.print("Discarded cards: ");

                while (card.size() - 1 >= 2) {
                    System.out.print(card.get(0) + ", ");
                    card.remove(0);
                    card.add(card.get(0));
                    card.remove(0);
                }
                System.out.print(card.get(0));
                card.remove(0);

                System.out.println("\nRemaining card: " + card.get(0));
            }
            number = input.nextInt();
        }
        input.close();
    }
}

File: BRACU ACM Student Chapter/Online Workshop on Programming Skills DAY 03 ( C++ ) BRACU Data Structure/G.cpp
#include<bits/stdc++.h>
using namespace std;

int main()
{
    int number;
    queue<int> card;

    while(cin>>number)
    {
        if(number==0) break;
        else
        {
            for(int i=0; i<number; i++) card.push(i+1);
            cout<<"Discarded cards:";
            int i = card.size();
            while(i!=1)
            {
                cout<<" "<<card.front();
                if(i>2) cout<<",";
                card.pop();
                i--;

                card.push(card.front());
                card.pop();
            }

            cout<<"\nRemaining card: "<<card.front()<<"\n";
        }
        card = queue<int>();
    }
    return 0;
}
